builtin/logical/transit/path_export.go

```diff
@@ -208,6 +218,10 @@ func keyEntryToECPrivateKey(k *keysutil.KeyEntry, curve elliptic.Curve) (string,
 		return "", errors.New("nil KeyEntry provided")
 	}
 
+	if k.IsPublicKeyImported() {
+		return "", nil
```

Comment:
You think we should? I assumed that exporting a key that wasn't imported isn't a error and should return an empty string.

Not related but I also think I forgot about "this" check for RSA keys.

Reply:
I guess I was thinking based on the expectations of someone using the export API today. It would be weird to get back a response with no actual key from it. 

We could return the public key from this API instead, I'd rather that personally than get nothing back. 

And yup you are correct, we will need to address the RSA key case here otherwise we will get a nil error.

Reply:
Public keys should now be exported if there's no private present on RSA and ECDSA keys.

Also, while working on this suggestion, I also realized that there were two `if` statements that would block the response from being empty ([this](https://github.com/Gabrielopesantos/vault/blob/4c2b3eef1d0fdbf65a5ec9b5f3f2c7403e2c3385/builtin/logical/transit/path_export.go#L137) and [this](https://github.com/Gabrielopesantos/vault/blob/4c2b3eef1d0fdbf65a5ec9b5f3f2c7403e2c3385/builtin/logical/transit/path_export.go#L105))

Also updated the tests. While working on them I was trying to print the output response of the export request but for some reason the response `Data` field was empty, is that expected or was I doing something wrong?

Reply:
> Also updated the tests. While working on them I was trying to print the output response of the export request but for some reason the response `Data` field was empty, is that expected or was I doing something wrong?

The two things that come to mind would be
1. Make sure the key on import has the `exportable` field set to true, you'd have a Data field with an `error` key
2. Verify the key name for the export is correct, you'd see a nil, nil response back from this case.

Since the majority of transit tests aren't using the logical API client, but going directly into the `HandleRequest` call on the backend, checking for err technically isn't enough as we might also have an error within the response field, if the handler returned something like `logical.ErrorResponse("an error"), nil`.